### ISSUE TITLE
Recent avoid_redundant_argument_values fix breaks redirecting generative constructor support

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#3850]
**Describe the issue**

Recent avoid_redundant_argument_values fix breaks redirecting generative constructor support, causing false positives and false negatives.

**To Reproduce**

```dart
class A {
  A({int? value}) : this._(value: value);
  A._({int? value = 2});
}
void f() {
  A(value: 2); // false positive
  A(value: null); // false negative
}
```

**Comment by lrhn** [comment]
Agree.
My guess is that the lint is treating a forwarding generative constructor the same as a forwarding factory constructor.

Forwarding generative constructors differ from forwarding factory constructors in how they treat parameters.
A forwarding factory constructor forwards the argument list as-is, including absence of arguments.
A forwarding generative constructor just calls the other constructor with a new argument list. The target constructor's parameter list should be unimportant to the lint.

If the unnamed constructor here had been `factory A({int? value}) = A._;`, the lint warnings would have been correct.

(I'm a little worried by the lint exposing the behavior of a private constructor through a public factory constructor, but since it's also the behavior of the public constructor, it's *kind-of* fair.)
